### DISCUSSION TITLE
Syntax error for Create Table prepared statement

### DISCUSSION THREAD

**Discussion by paul-uz** [sidorares/node-mysql2#2017] [Q&A]
I get an error for the following execute command

`const table = await connection.execute('CREATE TABLE IF NOT EXISTS ? (? INT, ? VARCHAR(255))', ['news', 'id', 'title']);
`

> Error: You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near '? (? INT, ? VARCHAR(255))' at line 1

If I hard code the table name etc, the statement is fine.

This is on Mysql 5.7 (on AWS Aurora RDS) but will probably change to the latest version of MySQL, esp if its a MySQL version issue....

**Comment by paul-uz** [comment]
Looks like I needed to wrap the table name `?` in backticks :/

**Comment by paul-uz** [comment]
So today I am trying again, this time, using back ticks, but on a new DB (Aurora Serverless MySQL) and the create table queyr is running, with no errors, but not creating the table.

This is the output

```
[
  ResultSetHeader {
    fieldCount: 0,
    affectedRows: 0,
    insertId: 0,
    info: '',
    serverStatus: 2,
    warningStatus: 1
  },
  undefined
]
```

**Comment by sidorares** [comment]
@paul-uz you can't use table name as a prepared statement parameter

**Comment by paul-uz** [comment]
Oh, why not?

I'm not seeing any mention of this in the docs.

**Comment by sidorares** [comment]
Its a mysql server / sql syntax restriction.
Can't find a good page in mysql docs ( maybe I'm missing something here https://dev.mysql.com/doc/refman/8.0/en/sql-prepared-statements.html )

also related: https://stackoverflow.com/questions/53233631/mysql-prepare-statement-is-it-possible-to-parametrize-column-name-or-function
